### PR TITLE
feat(hub-common): adds geometry to search results and new permission …

### PR DESCRIPTION
**Pull request by tannerjt**
…policy

affects: @esri/hub-common

Adds geometry to search results and new permission policy for viewing geometry in a new gallery
map layout.

Story: [#8503](https://devtopia.esri.com/dc/hub/issues/8503)

1. Description: Adds `geometry` to gallery search results.  Also adds new policy `hub:feature:gallery:map` that will be disabled by default and will be required to explicitly enabled with `?pe=hub:feature:gallery:map`.

1. Instructions for testing:

1. Closes Issues: #<number> (if appropriate)

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Review by abp6318 — APPROVED** [review]
This is so frickin cool to me. I'm very excited to see all of this live :)

Two questions below, otherwise it looks great!

File: packages/common/src/permissions/HubPermissionPolicies.ts [hunk 1]
```diff
@@ -95,6 +95,12 @@ const SystemPermissionPolicies: IPermissionPolicy[] = [
     availability: ["alpha"],
     environments: ["devext", "qaext"],
   },
+  {
+    // This is an experimental extension of the gallery to include `map layout`
+    // Similar to `hub:feature:privacy`, we want to deny permission unless explicity enabled with (?pe=hub:feature:gallery:map)
+    permission: "hub:feature:gallery:map",
+    environments: ["devext"],
```
[the remark this answers is not in the document]
Comment (tannerjt) [reply packages/common/src/permissions/HubPermissionPolicies.ts:102]: I was using this [comment](https://github.com/Esri/hub.js/blob/b9b8a59793310cfcb672c4c38e5ede56247bc8da/packages/common/src/permissions/HubPermissionPolicies.ts#L88-L90) for guidance.  The goal is to `deny` this permission even on QA by default, and make it only enabled by explicit `?pe= hub:feature:gallery:map`.

[the remark this answers is not in the document]
Comment (tannerjt) [reply packages/common/src/permissions/HubPermissionPolicies.ts:102]: CC @dbouwman to confirm I'm doing this correctly.

[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/common/src/permissions/HubPermissionPolicies.ts:102]: Yep - this looks good @tannerjt !

**Review by sonofflynn89 — APPROVED** [review]

**Pull request merged**